web/src/js/components/Modes/HTTPS.tsx

```diff
@@ -0,0 +1,14 @@
+import * as React from "react";
+import Mode, { ModeType } from "./Mode";
+
+export default function HTTPS() {
+    return (
+        <Mode
+            title="Explicit HTTP(S) Proxy"
+            description="You manually configure your client application or device to use an HTTP(S) proxy."
+            type={ModeType.REGULAR}
+        >
+            <p>Run HTTP/S Proxy</p>
+        </Mode>
```

Comment:
My hot take here is that we're erring too much on the side of reusing code here, which causes unnecessary complexity by adding too many layers of abstraction. There is a tradeoff here, and we shouldn't be afraid to have minor amounts of duplication.

This here could also just be like that:

```jsx
export default function Regular() {
    const dispatch = useAppDispatch(),
          active = useAppSelector((state) => state.modes.regular.active);
    return <div>
        <h4>Explicit HTTP(S) Proxy</h4>
        <p class="description">You manually configure your client application or device to use an HTTP(S) proxy.</p>
        <div>
            <label>
                <input type="checkbox" checked={active} onChange={() => dispatch(modeActions.toggleRegular())} />
                Run HTTP/S Proxy
            </label>
        </div>
    </div>
}
```

Yes this adds some duplication, but we now don't have to look at Mode, ModeType, ModeToggle, or MenuToggle. Things become much easier to reason about and test.

Reply:
I see, that makes sense! 👍 I thought it was generally best practice to reuse code and avoid duplication, which is why I tried to code it abstractly. I agree, however, that it turned out to be difficult to understand. So, ideally, we'll have a file (similar to the one above) for each mode, regardless of code duplication, right? Regarding the ducks, I still need to understand how they work because they still seem like black magic to me 😅